Core changes

MFlisar opened the pull request:
**Changes**

* Home: fixed time receiver unregistration
* SearchBar: init recycler view improved
* SearchBar Mode: added id to the enum => maybe the enum will be removed and replaced with a simple string... 
* Changed the core a little back => only settings management, gesture handling, dialogs, image loading and apps loading is extracted now; moved a little more code into the core for this
* added enums for icon position (desktop or dock) and some constants (Definitions.java)
* further adjusted setup and dependencies to fully support foreign image loader library like glide => I'm already successfully using glide in my app
* adapter now clear their items correctly for smoother adapter scrolling appearance
* popup label color added
* logger added
* notifyRemoveListeners corrected - it may be possible that apps are installed and deinstalled without updating the apps list in AppManager so we should not ignore the case where old list is not bigger than the new list + function moved to tool class
* the beta image caching function can be disabled in the settings => no files will be copied or read from internal cache directory so no more error logs in this case and feature can simply be enabled in the future
* fixed AppItemView image loading for async image loading
* code cleaned
* bugfix: drag & drop item from folder to desktop was showing the wrong view => this bug was introduced by my last refactoring
* popup background color added to settings
* dismiss popup on back press added

**Why did I change the setup class again**

I had some problems because my main idea was to implement all views myself because OLs views do not support as much customisation as my main app does. I had some problems so I decided to go a step back and try to use OLs views and only extract image loading. This means, the setup class is much cleaner and more clear now, but again I changed a lot... Sorry for this. The good thing is, I am successfully using the core in my own app with my own setup instance and glide as image async image loader so the current setup class proves to be useable and therefore makes sense now. Currently I'm successfully using it without any issues OL is not having as well...

gsantner commented on core/src/main/java/com/benny/openlauncher/core/util/SimpleIconProvider.java:
```diff
@@ -0,0 +1,103 @@
+        if (drawable != null) {
+            return drawable;
+        } else if (drawableResource > 0) {
+            return Setup.appContext().getResources().getDrawable(drawableResource);
```
will this also work if the image is an VectorDrawable?

MFlisar replied:
This is simple native android resource loading, vector drawables will work as before. If you add a glide loader you will have to take care of vector drawables because glide does not support them out of the box yet. Other libraries may not need this.

MFlisar commented:
This is simple native android resource loading, vector drawables will work as before. If you add a glide loader you will have to take care of vector drawables because glide does not support them out of the box yet. Other libraries may not need this.

MFlisar commented:
Added split up `DesktopOptionView`... Currently just a simple solution by using the technic that is currently used. Could be improved by removing the `RecyclerView` and using a `LinearLayout` with evenly distributed items which would look nicer. But at least all options are visible now at once and are not hidden anymore behind a scrollable view...

MFlisar commented:
* grouped settings in `SettingsManager`
* Popup `CardView` will now remove it's elevation if a (partly) transparent background color is defined to avoid weird shadow drawings of the `CardView` because `CardView` does not support elevation and transparent backgrounds at once
* margins for vertical drawer are now defined in `SettingsManager`
* IconSize split up in icon size for drawer, dock and desktop
* setting to reset searchbar's `RecyclerView` scroll position on opening search view added

**Following should be checked**

I have made following change and added a comment to it:
https://github.com/MFlisar/openlauncher/blob/master/core/src/main/java/com/benny/openlauncher/core/widget/SearchBar.java#L280

I see that OL does cut off the last item in the searchbar's `RecyclerView` but I can't find out why... Above change solves the problem but there must be a reason and it would be better to find it and use the correct correction value instead of the guessed one...

dkanada commented on core/src/main/java/com/benny/openlauncher/core/widget/SearchBar.java:
```diff
@@ -266,18 +268,17 @@ else if (item.getLabel().toLowerCase().contains(s.toLowerCase()))
-                recyclerParams.setMargins(0, marginTop, 0, 0);
-                // TODO: if we don't subtract the margin twice the last item is cut off...
-                recyclerParams.height = ((View)getParent()).getHeight() - 2 * marginTop;
+                searchRecycler.setPadding(0, marginTop, 0, 0);
```
searchRecycler.setPadding(0, marginTop, 0, marginBottom) where marginBottom is the bottom inset from the desktop should work.

dkanada replied:
As far as the keyboard goes, not sure that's a design error. With the keyboard displayed a user wants to search the list and so the keyboard should get focus over the apps. If we want to fix this issue I would say add an option to autoshow the keyboard on search.

MFlisar replied:
Now that I've read this again this makes sense and I tried it and it works :-) Still not sure why I can't replace ` int marginTop = Tool.dp2px(50, getContext()) + searchInput.getHeight();` with ` int marginTop = searchInput.getTop() + searchInput.getHeight();` as the `searchInput` and the `RecyclerView` are placed in the same container `FrameLayout`...

MFlisar replied:
Keyboard Overlap: could be simply solved by making the layout react to the keyboard with the activity flag `adjustResize` but that's anyways another issue... And I won't change that now as this will have an activity wide effect and may have other side effects

MFlisar commented:
Will solve following issue:
https://github.com/OpenLauncherTeam/openlauncher/issues/187

MFlisar commented:
* added settings for fast scroller color
* new settings for searchbar: optional switch button for changing between list and grid, define grid size, define label line number
* fixed searchbar text input position to not overlap any button
* updated library dependencies
* added google repo => support libraries can now be downloaded from there directly via gradle, no need to download them in sdk manager anymore

New settings are all initialised in OL to don't change current look

gsantner commented:
> added google repo

Please don't as F-Droid buildbot will block it

MFlisar commented:
Sure? I'm bit talking about adding any Google code to the app but just to at the repo source (like maven) to the repositories gradle will look at to find aar files and other dependencies...

gsantner commented:
Yes. The repo cannot be added. Only repos with published sources are accepted.

MFlisar commented:
This means I must download the support library and similar twice? Because f-droid compatible apps can't use the already downloaded data from the google() repo?

gsantner commented:
No why? You download the support libs to your local android sdk installation / local maven repo. And the google shouldn't be there

MFlisar commented:
I don't. I add google() like I add maven() to the repositories and gradle
will check if it can find any dependency at any of those places and will
download them. I'm not using the sdk anymore. Google has moved to provide
their libraries aar online instead of via downloading them in the sdk.

Gregor Santner <notifications@github.com> schrieb am Do., 13. Juli 2017,
19:36:

> No why? You download the support libs to your local android sdk
> installation / local maven repo. And the google shouldn't be there
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenLauncherTeam/openlauncher/pull/185#issuecomment-315148751>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACUyXloM2wKCH34FrOmai4X0su_JMEXyks5sNlWBgaJpZM4OSJPM>
> .
>

gsantner commented:
Take a look at <https://gitlab.com/fdroid/fdroidserver/merge_requests/275>. I can't tell when it gets ready. If `google()` is really just a convienient as told in <https://docs.gradle.org/current/release-notes.html> for `url 'https://maven.google.com'` than I think it will work. Another thing is that F-Droid might not be ready yet for latest gradle. But that will change.

MFlisar commented:
Ok understand now. Then I'll remove Google for now again...

MFlisar commented:
* removed google maven repo again
* added some global setup vars to root project
* solved the issue of leftover items after removing pages from the desktop or folders => should solve https://github.com/OpenLauncherTeam/openlauncher/issues/182

gsantner commented:
@MFlisar 
I added your proposed changes about supportlib in the latest commit. As API26 and supportlib-v26 are both out of beta I think it's good point to update. I will see how/if it works with F-Droid already, but I think the problems should be all sorted out there now.
